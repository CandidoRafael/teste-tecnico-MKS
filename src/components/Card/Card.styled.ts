import { styled  } from 'styled-components'

export const CardContainer = styled.section`
    box-shadow: rgba(50, 50, 105, 0.149) 0px 2px 5px 0px,
    rgba(0, 0, 0, 0.05) 0px 1px 1px 0px;
    border-radius: 8px;
    background-color: #fff;
    display: flex;
    flex-direction: column;
    align-items: center;
    
`

export const CardImage = styled.img`
    width: 13.2rem;
    height: 220px;
    object-fit: contain;
`

export const CardTitle = styled.article`
    display: flex;
    align-items: flex-start;
    gap: 5px;

    h1 {
        font-size: 16px;
        width: 124px;
        height: 38px;
        font-style: normal;
        font-weight: 400;
    }

    span {
        background-color: #373737;
        color: #fff;
        padding: .3rem;
        border-radius: 5px;
        font-size: 15px;
        font-weight: 700;
    }
`

export const CardDesc = styled.p`
    margin: .4rem 0 .6rem 0;
    font-weight: 300;
    font-size: 10px;
    padding: .5rem;
`

export const CardButton = styled.button`
    display: flex;
    align-items: center;
    cursor: pointer;
    justify-content: center;
    gap: 5px;
    width: 100%;
    border: none;
    background-color: var(--color-blue);
    height: 32px;
    color: #fff;
    font-size: 14px;
    font-style: normal;
    font-weight: 600;
    text-transform: uppercase;
    border-radius: 0 0 8px 8px;
    transition: .3s all linear;

    &:hover {
        background-color: #1e66d3;
    }

    &:disabled {
        cursor: not-allowed;
    }
`