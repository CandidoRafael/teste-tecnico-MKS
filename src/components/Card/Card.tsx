import { Product } from "../../types/Product"
import { TextLimit } from "../TextLimit/TextLimit"
import { CardButton, CardContainer, CardImage, CardTitle } from "./Card.styled"

import useStore from "../../store/store";
import ShoppingBagIcon from '@mui/icons-material/ShoppingBag';
import AddShoppingCartIcon from '@mui/icons-material/AddShoppingCart';

const Card = ({ product }: { product: Product }) => {

  const { setCart, cart } = useStore()
  
  return (
    <CardContainer key={product.id}>
      <CardImage src={product.photo} alt="imagem produto" />
      <CardTitle>
       <h1>{product.name}</h1>
       <span>R${product.price}</span>
      </CardTitle>
    
      <TextLimit text={product.description} limit={89} />

    
      <CardButton onClick={() => setCart(product)} disabled={cart.some((item) => item.id === product.id)}>
        {cart.some((item) => item.id === product.id) ? (
          <>
            <AddShoppingCartIcon />
            Adicionado
          </>
        ) : (
          <>
          <ShoppingBagIcon />
          Comprar
          </>
        )}
      </CardButton>

    </CardContainer>
  )
}

export default Card