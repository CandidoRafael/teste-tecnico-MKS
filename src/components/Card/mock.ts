export const cardPropsMock = {
    id: 1,
    name: 'Produto 1',
    brand: 'Marca Produto',
    description: 'Descrição Produto',
    price: 3000,
    photo: 'img/img.png',
    quantity: 2
}