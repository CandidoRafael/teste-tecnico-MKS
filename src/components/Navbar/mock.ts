export const useStoreMock = jest.fn().mockReturnValue({
  toggleSidebar: jest.fn(),
  cart: [],
});