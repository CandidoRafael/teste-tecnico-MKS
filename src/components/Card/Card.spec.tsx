import { render, screen } from "@testing-library/react"
import Card from "./Card"
import { cardPropsMock } from "./mock"

const mock = cardPropsMock

describe('<Card />', () => {
    it('should render Card correctly', () => {
        render(<Card product={mock} />)
        
       expect(screen.getByRole('img', { name: 'imagem produto' })).toHaveAttribute('src', 'img/img.png')
       expect(screen.getByRole('heading', { name: 'Produto 1' })).toBeInTheDocument()
       expect(screen.getByText('Descrição Produto')).toBeInTheDocument()
    })

    it('should match snapshot', () => {
       const { container } = render(<Card product={mock} />)
       expect(container.firstChild).toMatchSnapshot()
    })
})