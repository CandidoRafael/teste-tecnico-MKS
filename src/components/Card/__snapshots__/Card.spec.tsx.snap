// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> should match snapshot 1`] = `
<section
  class="sc-bdfCDU gkkTvX"
>
  <img
    alt="imagem produto"
    class="sc-gsTDqH jDjcfa"
    src="img/img.png"
  />
  <article
    class="sc-dlfmHC lHBgN"
  >
    <h1>
      Produto 1
    </h1>
    <span>
      R$
      3000
    </span>
  </article>
  <p
    class="sc-hKgKIp lAaGH"
  >
    Descrição Produto
  </p>
  <button
    class="sc-eCstZk fcNmUX"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="ShoppingBagIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M18 6h-2c0-2.21-1.79-4-4-4S8 3.79 8 6H6c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V8c0-1.1-.9-2-2-2zm-8 4c0 .55-.45 1-1 1s-1-.45-1-1V8h2v2zm2-6c1.1 0 2 .9 2 2h-4c0-1.1.9-2 2-2zm4 6c0 .55-.45 1-1 1s-1-.45-1-1V8h2v2z"
      />
    </svg>
    Comprar
  </button>
</section>
`;
